### PR TITLE
Arm64: always save int registers at higher addresses than float regs

### LINKED ISSUE #22457: [arm64] Always save int registers at higher addresses than float registers
Currently, floating-point registers are stored at higher addresses than integer registers in some frame types. It would be nice to make the frame types all the same, with int registers stored at higher addresses. See the code in `CodeGen::genSaveCalleeSavedRegistersHelp` and `CodeGen::genRestoreCalleeSavedRegistersHelp`

### PR DESCRIPTION
**Pull request by BruceForstall**
This is just to be consistent across all frame types, for simplicity.

Fixes #22457

### CODE REVIEW

**Comment by BruceForstall** [comment]
@dotnet-bot test Ubuntu16.04 arm64 Cross Checked no_tiered_compilation Build and Test
@dotnet-bot test Ubuntu16.04 arm64 Cross Checked jitstress1 Build and Test
@dotnet-bot test Ubuntu16.04 arm64 Cross Checked gcstress0xc Build and Test

**Comment by BruceForstall** [comment]
There are a few minor diffs due to a small missing optimization: https://github.com/dotnet/coreclr/issues/23102.

**Comment by BruceForstall** [comment]
@sandreenko @briansull PTAL
cc @dotnet/jit-contrib

File: src/jit/codegenarm64.cpp [hunk 1]
```diff
@@ -736,31 +730,25 @@ void CodeGen::genRestoreCalleeSavedRegistersHelp(regMaskTP regsToRestoreMask, in
...
 
-        if (maskRestoreRegsFloat != RBM_NONE)
-        {
-            // If there is any spDelta, it must be used here.
-            genRestoreCalleeSavedRegisterGroup(maskRestoreRegsFloat, spDelta, spOffset);
-            // No need to update spOffset since it's not used after this.
-        }
+    if (maskRestoreRegsInt != RBM_NONE)
+    {
+        int spIntDelta = (maskRestoreRegsFloat != RBM_NONE) ? 0 : spDelta; // should we delay the SP adjustment?
```
Comment (sandreenko) [inline src/jit/codegenarm64.cpp]: I do not understand that comment. Do you mean to do sp adjustment here like:
```
if (maskRestoreRegsInt != RBM_NONE)
{
  genRestoreCalleeSavedRegisterGroup(maskRestoreRegsInt, spDelta, spOffset);
  spDelta = 0;
}
if (maskRestoreRegsFloat != RBM_NONE)
{
  genRestoreCalleeSavedRegisterGroup(maskRestoreRegsFloat, spDelta, spOffset);
}

```

Is it for correctness or for readability?

Comment (BruceForstall) [reply src/jit/codegenarm64.cpp]: Note that I haven't changed this logic; I've just picked one path and un-indented it.

In this case, we need to do the opposite of the "save" case. In the "save" case, we make the spDelta adjustment with the first thing saved -- either the float registers, if any, or the int registers if no float regs. When restoring, we make the spDelta adjustment *last* -- so we need to see if there are any float regs that we'll restore later, and defer the spDelta adjustment if there are.

Comment (sandreenko) [reply src/jit/codegenarm64.cpp]: Got it, I misunderstood the comment.

**Comment by BruceForstall** [comment]
No altjit crossgen asm diffs in corelib after #23102 fixed.

**Pull request merged**